src/lib/upgradePackageDefinitions.ts

```diff
@@ -11,6 +11,77 @@ import { pickBy } from './pick'
 import queryVersions from './queryVersions'
 import upgradeDependencies from './upgradeDependencies'
 
+/**
+ * check if in peer violation
```

Comment:
Let's come up with a more descriptive JSDOC comment for this function. 

What does it mean for a peerDependency to be in violation? What are the different conditions that this function checks?

Reply:
Done